### PR TITLE
Include FAQ page as in-app help

### PR DESCRIPTION
**Pull request by pabzm**
Uses the faq-page of the website as content. The HTML has been changed
to only contain the div#content without the h1, and the path of
delta-what.png has been fixed to "./"

- ~~[ ] preprocess the HTML when building the app, not when serving the page.~~ (later)
- [x] write tests for the preprocessing
- [x] figure out how to serve localized content.

### CODE REVIEW

**Comment by pabzm** [comment]
Could someone give me a hint why the links in the dialog don't trigger opening the url in a browser? I'm a little clueless about that.

**Comment by pabzm** [comment]
Related to https://github.com/deltachat/deltachat-android/pull/1207

**Comment by Simon-Laux** [comment]
security, you need too loop through them and convert them like here:
https://github.com/deltachat/deltachat-desktop/blob/master/src/renderer/components/message/MessageMarkdown.js#L55-L59

**Comment by Jikstra** [comment]
Can we have a script which converts the faq from the deltachat-pages repo?

**Comment by r10s** [comment]
> Can we have a script which converts the faq from the deltachat-pages repo?

yip, the plan is to put that to deltachat-pages/tools/ - the script writes to ../deltachat-android/ ../deltachat-desktop and ../deltachat-ios then.

**Comment by Simon-Laux** [comment]
_when we already do a script the proper faq format in own repo is not far..._

**Comment by hpk42** [comment]
On Wed, Jan 08, 2020 at 21:23 -0800, Simon Laux wrote:
> _when we already do a script the proper faq format in own repo is not far..._

but this requires new workflows for translations -- IMO for january it's good 
to just keep pages working including established translations flows,
and extract the info into the other repos for inclusion in-app.

**Comment by r10s** [comment]
i've create a script that strips unneeded things from the faq at https://github.com/deltachat/deltachat-pages/pull/245 - the script is called from deltachat-pages and assumes deltachat-android and deltachat-desktop located beside deltachat-pages  - i would rename static/faq/faq.html to static/faq/html.html and then we can just add it to the python script.

**Comment by pabzm** [comment]
I'm reassigning myself until my part is finished

**Comment by pabzm** [comment]
Done, please review.

File: src/renderer/components/dialogs/HelpPage.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    <p style={{ textAlign: 'center', fontSize: 'smaller', borderTop: 'solid thin lightgrey', paddingTop: '0.3em' }}>
+      {linkedText}
+    </p>
+  )
+}
+
+export default function HelpPage (props) {
+  return (
+    <DeltaDialog
+      title={window.translate('help_page_title')}
```
Comment (Jikstra) [inline src/renderer/components/dialogs/HelpPage.js]: The menu item in german says 'Hilfe' but the dialog title is 'Frequently asked Questions'. I would like to have it consistent, but probably this is a translation fault?

**Comment by Jikstra** [comment]
What is this `.keep-dir` file?

File: src/renderer/components/dialogs/HelpPage.js (hunk 1 above)
Comment (r10s) [reply src/renderer/components/dialogs/HelpPage.js]: yes, i would also call it "Help" in the app everywhere.

**Comment by pabzm** [comment]
`.keep-dir` is a "hack" to make git not ignore empty directories. git tracks content, thus files, not file system entries like directories. In order to be able to run the script from deltachat-pages that produces the HTML-files we must have the directories in the repo. As long as we have a HTML-file in the repo, this "hack" isn't actually necessary, but I figured that it doesn't hurt either.

**Comment by Simon-Laux** [comment]
I saw `.gitkeep` already I few times, maybe that's a more common/standard name for it?

**Comment by pabzm** [comment]
To git the name of the file doesn't matter. If you prefer your suggestion feel free to rename it.

**Comment by Simon-Laux** [comment]
So just reviewed it with holger:
- Performance wise we don"t need to do anything its fine as is.
- The indentation of the links at the beginning is missing (probably some css).
- the FAQ should also be reachable over the window menubar "Help" menupoint (there where also the about button is).

Something worth considering but not urgent enough to do now:
- Open it in a separate electron window so you can read the FAQ while using the app - that would be super useful when following instructions on settings from the FAQ.
- the html tag causes an react error in the dev console (idea - maybe we could just an iframe that has a small script tag that converts the links for us?, that way the help.css would make more sense again)

**Comment by pabzm** [comment]
Parsing the HTML when building the app, instead of when the "help"-Link was clicked, would win two things:
1. Performance
2. The guarantee that a packages app contains parseable HTML.

Even if 1. is neglectable, 2. is still a valid point I think. But we can also add this check as a test.

**Comment by pabzm** [comment]
A separate window would be great! I just didn't know how to do it and wanted a relevantly useful feature first.

The react error is a warning, IIRC. I can live with that, if the alternative is ugly or expensive:
I tried shortly to use an iFrame, but then dropped it to find a easier solution. If the iFrame can't access it's resource via an URI we would need to write content into it using ugly hacks (stringify content, assign that as html of the iframe's document; I did it in a different project and still am ashamed ;).

And using a separate window would maybe solve the second issue, too, anyway?

**Pull request merged**